inc/SoundEmojiSynthesizer.h

```diff
@@ -128,7 +128,7 @@ namespace codal
         int                     samplesWritten;         // The number of samples written from the current sound effect block.
         float                   position;               // Position within the tonePrint.
         float                   samplesPerStep[EMOJI_SYNTHESIZER_TONE_EFFECTS];     // The number of samples to render per step for each effect.
-
+        bool                    stopping;               // Tracks when a stop is requested until it is acted on in a pull.
```

Comment:
I could use a status flag instead?

Also not sure whether all these fields should be public, but I've followed that pattern.

Reply:
I agree, a lot of these should be private really. Looks like an oversight. Let's fix that in a different commit. bool is fine, but optimising to a status flag would be nice to save a few bytes though :)

Reply:
Status flag done in https://github.com/lancaster-university/codal-microbit-v2/pull/10